### PR TITLE
Backport refmvs changes from dav1d 1.2.0

### PR DESCRIPTION
**Pull request by fbossen**
Make `refmvs_temporal_block` a packed structure and abstract functions for loading and saving it. Add asm functions for saving mvs. Partially addresses #225

### CODE REVIEW

File: src/decode.rs [hunk 1]
```diff
@@ -8255,6 +8251,7 @@ pub unsafe extern "C" fn dav1d_decode_tile_sbrow(t: *mut Dav1dTaskContext) -> li
         && (*(*f).frame_hdr).frame_type as libc::c_uint & 1 as libc::c_uint != 0
     {
         dav1d_refmvs_save_tmvs(
+            &(*((*f).c)).refmvs_dsp,
```
Comment (kkysen) [inline src/decode.rs:8254]:
```suggestion
            &(*(*f).c).refmvs_dsp,
```

File: src/refmvs.c [hunk 2]
```diff
@@ -794,8 +786,10 @@ void dav1d_refmvs_save_tmvs(const refmvs_tile *const rt,
                     rp[x] = (refmvs_temporal_block) { .mv = cand_b->mv.mv[0],
                                                       .ref = cand_b->ref.ref[0] };
             } else {
-                for (int n = 0; n < bw8; n++, x++)
+                for (int n = 0; n < bw8; n++, x++) {
+                    rp[x].mv.n = 0;
                     rp[x].ref = 0; // "invalid"
+                }
```
Comment (kkysen) [inline src/refmvs.c:792]: Was this change re-transpiled?

**Review by kkysen — CHANGES_REQUESTED** [review]
See the above comments.  The `rp[x].mv.n = 0;` change I believe was missed in the manual re-transpile.

File: src/refmvs.rs [hunk 3]
```diff
@@ -433,7 +487,8 @@ unsafe fn add_temporal_candidate(
     r#ref: refmvs_refpair,
     globalmv: Option<(&mut libc::c_int, &[mv; 2])>,
 ) {
-    if rb.mv == mv::INVALID {
+    let rb_mv = rb.mv;
+    if rb_mv == mv::INVALID {
```
[the remark this answers is not in the document]
Comment (fbossen) [reply src/refmvs.rs]: Thanks for pointing this out. Changed as suggested

File: src/refmvs.c (hunk 2 above)
Comment (fbossen) [reply src/refmvs.c:792]: This was missing indeed. Thanks for pointing it out. Added initialization of `mv`

**Review by kkysen — APPROVED** [review]

**Pull request merged**